src/Math/include/BipedalLocomotion/Math/AlgebraicRiccatiEquation.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file AlgebraicRiccatiEquation.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_MATH_ALGEBRAIC_RICCATI_EQUATION_H
+#define BIPEDAL_LOCOMOTION_MATH_ALGEBRAIC_RICCATI_EQUATION_H
+
+#include <utility>
+
+#include <Eigen/Dense>
+
+namespace BipedalLocomotion
+{
+namespace Math
+{
+/*
+ * Computes the unique stabilizing solution S to the continuous-time algebraic equation
+ * @f[
+ *   S A + A' S - S B R^{-1} B' S + Q = 0
+ * @f]
+ *
+ * @param A n x n square matrix
+ * @param B n x m matrix
+ * @param Q n x n symmetric-square matrix
+ * @param R m x m square positive definite matrix
+ * @param tolerance max admissible error
+ * @param maxIterations max number of iterations
+ * @param verbosity level
+ * @return a pair. If the Function is able to find a solution the first term is true and the second
+ * one contains the solution. Otherwise the first element of the pair is false.
+ * @note This function implements the algorithm presented in the paper: Solving the algebraic
+ * Riccati equation with the matrix sign function
+ * https://www.sciencedirect.com/science/article/pii/0024379587902229
+ */
+std::pair<bool, Eigen::MatrixXd>
```

Comment:
It is a pity to return a matrix, especially considering that this method can be called in controllers that need to run fast. I would suggest making it an input and return a simple boolean.

Reply:
I would keep both the implementation. For instance, in an LQR this matrix is computed only once

Reply:
Well, in an infinite-time LQR is computed only once :wink: In any case, I agree, it is possible to have both versions. As mentioned [below](https://github.com/dic-iit/bipedal-locomotion-framework/pull/157#discussion_r531680512) you could define a static method and/or exploit overloading